### PR TITLE
Allow flash/compile to accept relative json paths

### PR DESCRIPTION
**Pull request by zvecr**
## Description

Common issue that gets reported in the Discord, where the initial `cd qmk_firmware` within cli changes the assumption of where the relative path is to.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: lib/python/qmk/path.py [hunk 1]
```diff
@@ -65,3 +66,11 @@ def normpath(path):
...
 
     return Path(os.environ['ORIG_CWD']) / path
+
+
+class FileType(_FileType):
+    def __call__(self, string):
+        maybe_relative = Path(os.environ['ORIG_CWD'], string)
+        if maybe_relative.exists():
+            string = maybe_relative.resolve()
+        return super().__call__(string)
```
Comment (skullydazed) [inline lib/python/qmk/path.py:76]:
```suggestion
        return super().__call__(normpath(string))
```

File: lib/python/qmk/path.py [hunk 2]
```diff
@@ -65,3 +66,11 @@ def normpath(path):
         return path
 
     return Path(os.environ['ORIG_CWD']) / path
+
+
+class FileType(_FileType):
```
Comment (skullydazed) [inline lib/python/qmk/path.py:71]:
```suggestion
class FileType(argparse.FileType):
```

File: lib/python/qmk/path.py [hunk 3]
```diff
@@ -3,6 +3,7 @@
 import logging
 import os
 from pathlib import Path
+from argparse import FileType as _FileType
```
Comment (skullydazed) [inline lib/python/qmk/path.py:6]: Github won't let me suggest it, but this should go on line 3 as well to keep everything nicely ordered.

```suggestion
import argparse
```

**Review by skullydazed — CHANGES_REQUESTED** [review]

File: lib/python/qmk/path.py (hunk 1 above)
Comment (zvecr) [reply lib/python/qmk/path.py:76]: Applying this suggestion would change the existing behaviour of allowing a relative file path to the qmk_firmware directory.

Comment (skullydazed) [reply lib/python/qmk/path.py]: I'd rather break advice given to work around a bug than continue to support that behavior. Users shouldn't care that internally `qmk` is operating from a different current directory.

**Review by Erovia — APPROVED** [review]

**Pull request merged**